src/lock.c

```diff
@@ -114,17 +114,20 @@ do_lock_unlock(struct pkgdb *db, int match, const char *pkgname,
 		goto cleanup;
 	}
 
-	while ((retcode = pkgdb_it_next(it, &pkg, 0)) == EPKG_OK) {
-		if (action == LOCK)
-			retcode = do_lock(db, pkg);
-		else
-			retcode = do_unlock(db, pkg);
-
-		if (retcode != EPKG_OK) {
-			exitcode = EXIT_FAILURE;
-			goto cleanup;
-		}
-	}
+	if ((retcode = pkgdb_it_next(it, &pkg, PKG_LOAD_BASIC)) == EPKG_OK) {
+		do {
+			if (action == LOCK)
+				retcode = do_lock(db, pkg);
+			else
+				retcode = do_unlock(db, pkg);
+
+			if (retcode != EPKG_OK) {
+				exitcode = EXIT_FAILURE;
+				goto cleanup;
+			}
+		} while ((retcode = pkgdb_it_next(it, &pkg, PKG_LOAD_BASIC)) == EPKG_OK);
+	} else
+		printf("%s: package not found\n", pkgname);
 
```

Comment:
how is it that this error can only happen once?

Reply:
It occurs once for every package that is not found. Please let me know if I had to do it differently. Thanks.

Reply:
It occurs once for every package that is not found. Please let me know if I had to do it differently. Thanks.